### ISSUE TITLE
Tiedtke updraft: the cloud-base zlift bonus also relaxes the cloud-top criterion

### ISSUE THREAD

**Issue by duncanwp** [climate-analytics-lab/jax-gcm#691]
Raised in review of #690 (Codex, P2). Parent: #614.

## What's wrong

`updraft.py` adds the cloud-base sub-grid buoyancy excess `zlift` to the updraft's termination test at **every** ascent level:

```python
surv_buoy = jax.nn.sigmoid((buoy_new + zlift_buoy) / w_term_buoy)
```

`zlift` defaults to 1 K (`zlift_buoy ≈ 0.033 m/s²`) against a termination sigmoid width of `smooth_term_buoy = 3e-4 m/s² ≈ 0.01 K`, i.e. a ~100x allowance. So the plume stays essentially fully active past its equilibrium level until its virtual-temperature deficit exceeds ~1 K. On a profile that is only weakly stable above the true cloud top, that means overly deep mass flux, detrainment and precipitation.

ECHAM gates its own ascent bonus on `klab == 1` (`mo_cuascent.f90:449`) — only where the level below is still sub-cloud — so the reference never relaxes the cloud-top criterion this way.

## Why it was not fixed in #690

`zlift` has to be in the termination test *somewhere*, or the cloud-base gate admits a base (`zbuo + zlift > 0`) that the very next ascent step rejects (`zbuo < 0`) and the plume dies one level above its own base — that is the failure #690 exists to fix.

The obvious restriction — latch the bonus off once the plume first achieves genuine buoyancy (`zbuoy_accum > 0`, which is where ECHAM's `klab` would flip) — was implemented and **rejected on measurement**:

| composed RCE column, day 2 | bonus throughout | latched off after first buoyant level |
|---|---|---|
| mean `P_conv` | 2.52e-5 kg/m²/s | **6.38e-6** (4x less) |
| mean-relative water closure | 0.71 % | **1.98 %** (worse than the 1.80 % before #690) |

A plume crosses more than one thin inhibition layer; a latch that trips on the first marginally-buoyant level kills it at the second.

## Why it is currently masked

The relaxed criterion does not bind today because the cloud top in this implementation is set by **organized detrainment and the scan ceiling, not by the buoyancy test** — which is #669 ("Detrainment, not buoyancy, sets the cloud top"). Probes on a hydrostatic tropical column, on a column deliberately built weakly stable above its equilibrium level, and on a near-undilute plume (`entrpen = entrscv = entrmid = 1e-6`) all terminate on the mass-flux floor with buoyancy still strongly positive (+1.4 m/s²). No constructed column changed its answer with or without the bonus.

**So this is blocked on #669.** Once detrainment is in the metre-based capped form and buoyancy actually governs the cloud top, this bonus becomes load-bearing and the defect becomes reachable.

## What to do then

Options, in rough order of preference:

1. Restrict the bonus by *depth above cloud base* rather than by buoyancy history — the scheme already has the `zdnoprc` pressure-depth-from-base concept for the precip zone, so an analogous "inhibition zone" depth is idiomatic and does not trip on a single marginal level.
2. Port `klab` properly, together with the sub-cloud walk in #684, so the ECHAM gating can be reproduced exactly.
3. Shrink `zlift` in the termination test only (e.g. use `cu_cminbuoy` there and the full `zlift` at the base), bounding the cloud-top relaxation to 0.2 K.

Whichever is chosen, re-run the composed water closure (`composed_budget_test`) and the RCE flicker bound — both moved measurably when this was last touched.

## Refs

Cross-referenced from the comment at the `surv_buoy` site in `updraft.py`. Related: #669 (blocker), #684 (`klab` walk), #683 (thvsig), #682 (trigger retuning).

**Comment by duncanwp** [comment]
Fixed on `fix/tiedtke-echam-faithful` in 6e54f22, completed in 13d3e0c.

Resolution is the one this issue anticipated: the ascent zlift bonus was deleted from the `cubase` path entirely — ECHAM adds it only where `klab(jk+1)==1` (mo_cuascent.f90:449), and `cubase` sets `klab(kcbot)=2`, so a surface-triggered plume can never qualify and the ~1 K relaxation of the cloud-top criterion this issue reported was pure invention. 13d3e0c then restored the bonus at its ONE legitimate site: the first ascent step above a `cubasmc` (klab==1) mid-level base, which became reachable when #697 landed the real mid-level trigger.

Closes when the branch merges.
